Feature/pi/csv to drive

youssef-m-ali opened the pull request.

kmyali committed:
Create README.md

kmyali committed:
Add files via upload

youssef-m-ali committed:
Delete README.md

colton-smith commented:
Initial comment - every pull request should have some sort of description, even if it is very brief. This just helps summarize the PR, sometimes people reviewing your code wont be very familiar with the subject matter.

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/client_secret.json:
```diff
@@ -0,0 +1 @@
+{"installed":{"client_id":"817401415477-3u8dtd999mmjgfd8uh30s61thsjl9l5b.apps.googleusercontent.com","project_id":"quickstart-1588226457658","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://oauth2.googleapis.com/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"Kd4RL8-NUgotTkDgv3aa3nWm","redirect_uris":["urn:ietf:wg:oauth ...
```
Is this your personal client id and token? May not want to have this on our Github as this is a public repository, I think it would be better if we generate and add this file during testing, and not have it pushed to the repo

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/drive_upload.py:
```diff
@@ -0,0 +1,27 @@
+from httplib2 import Http
+from oauth2client import file, client, tools
+
+def uploadCsv(filePath):
```
Have you thought about being able to pass the filepath as a command line argument? Maybe what you could have is an `if __name__ == '__main__'` Which passes the first command line argument to the `filePath` parameter, and maybe you could have it so that the client_secret.json file gets passed as a second argument to the function, that way it can also be specified as a command line argument. The reason it would be good to have this feature is so that we can also manually upload files from the command line, instead of just invoking this function in other python files.

We can chat about this today

colton-smith replied:
That's a sick idea and would make it a full on tool, but at the same time manual uploads can also be done via google drive.

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/drive_upload.py:
```diff
@@ -0,0 +1,27 @@
+    store = file.Storage('credentials.json') 
+    creds = store.get()
+
+    if not creds or creds.invalid:
```
I like how you check for validity, this is good practice.

Burke-Daniel reviewed (changes requested):
Great job on this @kmyali @youssef-m-ali , just a couple of comments and changes to address. Overall one thing I notice is that there are a lack of comments throughout the function. Some of the stuff that is going on in the function is a bit hard to understand without a knowledge of the drive api, it wouldn't be a bad idea to add in some descriptive comments where you feel the code is not self explanatory. Also, might be nice to add a docstring to this function, see my pull request for the influxdb class to see how to structure docstrings for functions.

colton-smith commented on Pidaq/Lib/uploadToDrive/drive_upload.py:
```diff
@@ -0,0 +1,27 @@
+        
+    DRIVE = build('drive', 'v3', http=creds.authorize(Http()))
+
+    metadata = {'name': 'testData.csv',
+                'mimeType' : 'application/vnd.google-apps.spreadsheet',
+                'parents' : ['1CO4KyW5FHYHWAbz2JOh6XeDRzaY9I1WF']
+                }
```
Does this mean that the file will have the name testData.csv in google drive? It would be good if you could either specify the filename or just get it from the path (`os.path.basename()`)

Burke-Daniel replied:
Super minor style thing, a better way to format this dict would be:
```python
metadata = {
                'name': 'testData.csv',
                'mimeType' : 'application/vnd.google-apps.spreadsheet',
                'parents' : ['1CO4KyW5FHYHWAbz2JOh6XeDRzaY9I1WF']
           }
```

colton-smith commented on Pidaq/Lib/uploadToDrive/drive_upload.py:
```diff
@@ -0,0 +1,27 @@
+    res = DRIVE.files().create(body=metadata, media_body=filePath).execute()
+
+    if res:
+        print('Uploaded "%s" (%s)' % (metadata['name'], res['mimeType']))
```
Once the logging module is merged, you can log this as opposed to printing it

colton-smith commented on Pidaq/Lib/uploadToDrive/drive_upload.py:
```diff
@@ -0,0 +1,27 @@
+from googleapiclient.discovery import build
+from httplib2 import Http
+from oauth2client import file, client, tools
+
```
This is being very picky, but all top level functions and classes should be surrounded by 2 lines :)

Burke-Daniel commented:
Gitignore the client_secret and credentials json files in the top level gitignore

colton-smith reviewed (changes requested):
This is very promising guys, great job!

Couple of main changes needed:
- Need a PR description
- Really need a readme describing the process of using the token, selecting a drive folder etc etc
- Docstrings

youssef-m-ali committed:
Add files via upload

colton-smith commented on Pidaq/Lib/uploadToDrive/RunInCmd.py:
```diff
@@ -0,0 +1,14 @@
+import uploadToDrive
+
+path = input("File Path:") #Takes the file's path without quotation marks
+
```
Check if the path exists, if it doesn't - ask them to enter a valid path.

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,121 @@
+DRIVE = build('drive', 'v3', http=creds.authorize(Http()))
+
+
+
```
This should only be a single line space

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,121 @@
+    #Dictionary for parent folders and their IDs.
+    #These IDs are to be stored manually.
+
+    
```
Should only be one line of space here

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,121 @@
+
+def uploadCsv(filePath, parent):
+    """
+    Uploads a csv file from a local directory to a folder on Google Drive.
```
This line should follow the """, like:
""" Uploads a csv file from a local dir....

"""

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,121 @@
+    The parent folder is supposedly going to be fixed for a
+    long period of time.
+    """
+    parentDictionary = {'Safety':'1Vnb9UOTatdZUpcBXr1LBjziwNYZzRXQz',
+                        'Integrity': '1Kyr_bJ5MSN3b6w1KMZzdJ7YPyTgPxze1',
+                        'DTS': '100OsZP0JQIk9_VXHXpK8mdhS9GOp3yKm'}
```
Few things about this

* This dictionary should be moved to a config json file (name it what you want). 
* Read in the dictionary from the json file
* Explain how to find and add the folder keys to the file
* Wrap the parentFolder = parentDictionary[parent] in a try except statement to catch key errors (in case the user enters a folder name that isnt present in the dictionary). This will look like

```
try:
     parentFolder = parentDictionary[parent]
except KeyError:
    logging.error("some useful error message here")
    raise KeyError

```  
* Also check that the config file is present - if it is not, raise an error

youssef-m-ali replied:
I was thinking of putting parentFolder = parentDictionary[parent] in a while loop that checks if the folder name entered is in the keys of the dictionary. and if not, it would ask again. I believe it would be faster than raising an error and running the script from the beginning again. what do you think about that?

colton-smith replied:
Daniel: Instead of raising an error - use a default parent folder

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,121 @@
+                        'Integrity': '1Kyr_bJ5MSN3b6w1KMZzdJ7YPyTgPxze1',
+                        'DTS': '100OsZP0JQIk9_VXHXpK8mdhS9GOp3yKm'}
+    parentFolder = parentDictionary[parent]
+    #Dictionary for parent folders and their IDs.
```
Always put a space after single line comments for cleanliness 
```
# Hello
```

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,121 @@
+            }
+        folder = DRIVE.files().create(body = folder_metadata,
+                                      fields = 'id').execute()
+        #Creates a folder with the given date
+        folderId = getFolderId(date, parentFolder)
+        #Returns the date folder's ID
```
You should put comments above the code that they refer to. Like in a report always talk about a figure before the figure appears:

Instead of what you have, this would be cleaner
```
# Returns the date folder's ID
 folderId = getFolderId(date, parentFolder)
```

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,121 @@
+    #Today's date in YYYY-MM-DD format is stored in the variable "date"
+   
+    if checkExistingSubFolder(date, parentFolder) == 0:
+        #If the folder does not exist, should be the case
+        #when running for the first time on a given day
```
put this comment above the if statement

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,121 @@
+
+        
+def checkExistingSubFolder(folderName, parentFolder):
+    '''
```
All doc strings should use """

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,121 @@
+                'parents' : [folderId]
+                }
+
+    response = DRIVE.files().create(body=metadata,
+                               media_body=filePath).execute()
+    #uploads the .csv file
+
+    if response:
+        logging.warning('Uploaded "%s" (%s)' % (metadata['name'], response['mimeType']))
+    else:
+        logging.warning('error in uploading the csv file')
```
Line spacing should be used to outline a clear order of operations, so bundle things together that belong together as seen below:

```
    #uploads the .csv file
    response = DRIVE.files().create(body=metadata,
                               media_body=filePath).execute()

    if response:
        logging.warning('Uploaded "%s" (%s)' % (metadata['name'], response['mimeType']))
    else:
        logging.warning('error in uploading the csv file')
```

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,121 @@
+
+    if len(response.get('files',[])) == 0: 
+        logging.warning("creating a new folder...")
+        return 0
```
Its a little strange to return 0 and 1 here, is there any reason you did so as opposed to just returning True or False?

youssef-m-ali replied:
I didn't think there would be any difference. I just returned the number because it represents the number of folders with that date. Is it preferred to use True and False instead?

colton-smith reviewed (changes requested):
Good job guys, functionality is great - just needs a few style changes, error handling, and a readme.

youssef-m-ali committed:
Updating UploadToDrive.py

- I wasn't sure which error to raise if the config json file does not exist, so I decided to just exit the program.
- Should the ID of the Default folder be in the json file as well? I put it as a variable in the script.

youssef-m-ali committed:
Update uploadToDrive.py

youssef-m-ali committed:
Update RunInCmd.py

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,125 @@
+        parentDictionary = json.load(json_file)
+except:
+    logging.error('Parents dictionary .json file not found')
+    exit()
```
- Its good practice to specify what error you are looking for like ` except FileNotFoundError:` instead of using an all encompassing try except
- exit() is preferably only used in python shells as opposed to scripts / functions. Instead use sys.exit() or raise SystemExit. Although - as seen below i'm not sure if this is what we want to do in that situation.
- I think this check should occur either in an init() function or in one of the already defined functions. Otherwise, when you import this function with no dictionary.json in the cwd (we should be able to specify the location as well) - the program will exit. Instead, if there is no `dictionary.json` file, the function returns -1 or 1 or something to indicate a failure. 
- `dictionary.json` isnt very descriptive, maybe something along the lines of `drive_options.json` could be used

colton-smith replied:
Remember - the purpose of this script is to allow us to simply call a function and upload a file - we don't want to do too much in a "global" scope if you know what I mean.

Burke-Daniel replied:
Yeah 100% agree with all this, I don't see anything stopping you guys from moving this stuff to an init() function, and then specifying that this has to be ran the first time before the other functions are called. This makes the module much more portable, meaning we don't have to directly run this file in order to initialize the api functionality, it can all be done within one of our main programs.

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,125 @@
+except:
+    logging.error('Parents dictionary .json file not found')
+    exit()
+# change this for the ID of the default folder 
```
Very minor - I just think that we have been using capitalized first letters in comments. Not a big deal at all - we just want to be consistent as much as we can.

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,125 @@
+        parentFolderID = DefaultID
+        logging.error('Folder name not found \nuploading to "Default"')
+    
+    #Today's date in YYYY-MM-DD format is stored in the variable "date"
```
\# T

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,125 @@
+    elif checkExistingSubFolder(date, parentFolderID) == 1:
+        logging.warning("found folder for today's date")
+        folderId = getFolderId(date, parentFolderID)
+    # executes if more than one folder exists
+    else:
+        logging.warning('unexpected count of folders named with a given date')
+        return
```
Does google drive allow 2 folders with the exact same name in the same location?

youssef-m-ali replied:
Yes and the IDs will be different

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,125 @@
+        
+def checkExistingSubFolder(folderName, parentFolder):
+    """ Queries the drive for the folder with
+        the given name and parent folder
```
Again being picky - but there should not be a tab in the second line:

````
""" Queries the drive for the folder with
the given name and parent folder
"""

Burke-Daniel replied:
This statement could probably just fit on one line to be honest, general rule to follow is to keep your lines under 80 characters.

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,125 @@
+
+def getFolderId(fName, parentFolder):
+    """ Performs the query to return the ID of the folder
+        with the given name
```
See above comment about docstring spacing

Burke-Daniel replied:
Ditto

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,125 @@
+    long period of time.
+    """
+    # searches for the parent name in the dictionary
+    for key in parentDictionary.keys():
+        if key.lower() == parentName.lower():
+            parentKey = key
+    try:
+        parentFolderID = parentDictionary[parentKey]
+    except KeyError:
+        # uploads to a default folder if the folder name does
+        # not exist in the dictionary
+        parentFolderID = DefaultID
+        logging.error('Folder name not found \nuploading to "Default"')
```
2 things:
1.) In the first for loop - you would generally break after you find what you are looking for. Also - in the case where the key is not in the dictionary - `parentKey` never gets defined and you will get an error you arent looking for when you try `parentDictionary[parentKey]`
2.) Its a little weird how you iterate through all the keys first, looking for a matching key. And then afterwards - you check again if it exists (and in this case if the key doesnt exist - `parentKey` doesnt get initialized and an error will occur)

Couple of solutions here - not sure what we want to do for sure.

* Instead of iterating through all the keys and fixing the case there - when you load in the dict you could .lower() all the keys like:
```
json_dict = json.load()
parentDictionary = {}
for key, value in json_dict.items():
    parentDictionary[key.lower()] = value
```
 and then just try `parentDictionary[parentKey.lower()]`. This way we the json can contain be either uppercase or lowercase folder names as keys, and we will still find it when we do a lookup.

This is probably not the best solution as we are copying a whole dict for not much reason.

* We could simply say all folder names have to be all lower case or all upper case in the json, and just do the corresponding `.lower()` or `.upper()` to the key that we pass to the function.

* Finally, it really isn't that deep. We could also just not worry about case - if the key isn't there exactly as you pass it, an error is thrown and you fix your call to the function - easy as that. The important thing is you catch the case where there is no corresponding folder.

Burke-Daniel replied:
Yeah I think one of the solutions Colton suggested is suitable for this. In this case not too worried about performance for this, since none of this will run during a test.

youssef-m-ali replied:
yeah, it should have been NameError as `parentKey` would not be defined in this case. I think the way Colton is suggesting would be better. All upper or all lower might be something we want to specify in the README that's about to get uploaded, but we wouldn't have to if this works

colton-smith reviewed (changes requested):
Sorry to request more changes - but there is some vital changes that have to be made this time ( and a few more small style things)

HOWEVER, the style is significantly improved over the previous iteration - really nice job with that!!!

Let me know if you have any questions about my comments - a few in particular opened up a small rabbit hole about how we want this thing to function.

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,125 @@
+    """ Queries the drive for the folder with
+        the given name and parent folder
+    """
+    query = " name = '" + folderName + "' and parents = \"" + parentFolder +'"'
```
One thing to make this a little bit easier to read would be to use an [f-string](https://realpython.com/python-f-strings/), just improves readability a bit. There is also the `str.format()` method which is just as good.

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,125 @@
+from __future__ import print_function
+import os
+import logging
+from datetime import datetime
+import json
```
I think what we decided on for ordering these imports is to put the `import ...` statements first, in alphabetical order, followed by the `from ... import ...` statements, also in alphabetical order.

youssef-m-ali replied:
I will fix that. Regarding the `parentDictionary`, I initially had it inside the function but decided to take it out because I am calling it from outside the file in RunInCmd.py. I edited the code to display the keys in the dictionary (to show the folders available) to whoever is uploading.

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,125 @@
+DefaultID = '1VDOOjuOeyNRJgdKmmRTJC1HVjq91F4Q8'
+
+def uploadCsv(filePath, parentName):
+    """ Uploads a csv file from a local directory to a folder on Google Drive.
+
+    Parameters:
+    "filePath" is the path of the csv file itself and
+                should be passed without quotations.
+    "parentName" is the name of the testing folder in the drive
+            and only a specific set of values are allowed.
+    The parent folder is supposedly going to be fixed for a
+    long period of time.
+    """
```
This is a great lookin docstring!

youssef-m-ali replied:
Well, That's @kmyali's work. Thanks Kamal for writing the only docstring that looks good :D

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,125 @@
+        return
+    
+    # extracts the .csv file name from the path
+    fileName=os.path.basename(filePath)
```
*`filename = ...`

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,125 @@
+    
+    # extracts the .csv file name from the path
+    fileName=os.path.basename(filePath)
+    metadata = {'name': fileName,
+                'parents' : [folderId]
+                }
```
The way you formatted the dict on lines 62-66 looked a bit neater

youssef-m-ali replied:
I suggest making this dict in one line. The other one has 3 items and is longer. This one originally had 3 before we decided to remove the item that caused an error on one of the laptops.

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,125 @@
+    # Returns 1 if the folder with today's date
+    # exists in the specified parent folder
+    elif len(response.get('files',[])) == 1: 
+        for data in response.get('files', []):
```
What does this line do? Couldn't you just return 1 if that condition evaluates to True? The for loop doesn't seem to be doing anything.

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,125 @@
+        with the given name
+    """
+    # searches in the drive using the parameters passed
+    query = " name = '" + fName + "' and parents = \"" + parentFolder +'"'
```
See comment on line 99.

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,125 @@
+    response = DRIVE.files().list(q = query).execute()
+    # Gets the folder ID if it exists
+    if len(response.get('files',[])) == 1:
+        for data in response.get('files', []):
```
Is this loop necessary? you check that the length is equal to 1, so there is nothing to loop through. This could just be something like:
```python
return response.get('files', []).get('id')
```

Burke-Daniel reviewed (changes requested):
A few more questions/comments, I think we should have a chat about this either some day this week or at the meeting this weekend. I see that you are utilizing the `parentDictionary` variable in the `uploadCsv` function, thus there is a possibility that this variable won't be initialized. If that could be moved to inside the `uploadCsv` function or returned from the `init()` function where it could then be passed in as a parameter, or you could put these functions in a class, so you could then store that information as a member variable. Anyway, all this will be easier to explain on a call haha, but great job guys! This code is looking very neat and well documented, with a few structure changes it should be good to go!

youssef-m-ali committed:
Update uploadToDrive.py

youssef-m-ali committed:
Update uploadToDrive.py

youssef-m-ali committed:
Update uploadToDrive.py

youssef-m-ali committed:
Add files via upload

youssef-m-ali committed:
Delete drive_upload.py

Deleted the old file

youssef-m-ali committed:
Update uploadToDrive.py

- the if statement in getFolderId() is not necessary
- updated the .json file name
-added the FileNotFoundError

colton-smith commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,118 @@
+        logging.warning('error in uploading the csv file')
+        
+def checkExistingSubFolder(folderName, parentFolder):
+    """ Queries the drive for the folder with the given name and parent folder
+    """
```
One line docstrings have the quotations on the same line 

""" XXXXXXXXXXXXXXX """

youssef-m-ali committed:
Add files via upload

Grouped the functions under a class
moved the variables from the global scope to an __init__()
The script now creates a JSON file with the default ID if the file does not exist
solved the .lower() issue in the way Colton had in his comment
- will add more comments and the README in a day or two

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/RunInCmd.py:
```diff
@@ -0,0 +1,15 @@
+folder = input("Folder Name: ")
+
+uploadToDrive.uploadCsv(path, folder)
+
```
Change this to work with the class instead of the functions

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/driveClass.py:
```diff
@@ -0,0 +1,127 @@
+from httplib2 import Http
+from oauth2client import file, client, tools
+
+class uploadCsv:
```
In python, convention is for class names to be UpperCamelCase, so this would be UploadCsv

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/driveClass.py:
```diff
@@ -0,0 +1,127 @@
+                self.parentDictionary = json.load(json_file)
+                
+        except FileNotFoundError:
+            logging.error('Creating a drive options dictionary file...')
```
I think it might be more appropriate for this to be a logging.warning, which states "Json file could not be found, creating a default drive options dictionary file..."

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/driveClass.py:
```diff
@@ -0,0 +1,127 @@
+                json.dump(newDictionary, new_file)
+    
+            with open('drive_options.json') as json_file:
+                self.parentDictionary = json.load(json_file)
```
This is not necessary, you can instead just say `self.parentDictionary = newDictionary`

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/driveClass.py:
```diff
@@ -0,0 +1,127 @@
+        "filePath" is the path of the csv file itself and
+                    should be passed without quotations.
+        "parentName" is the name of the testing folder in the drive
+                and only a specific set of values are allowed.
```
A nicer style for this might be:
```python
"""
...
Parameters:
    filePath: path of the csv...
    parentName: ...
...

This is just me being picky, and totally not necessary haha
"""
```

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/driveClass.py:
```diff
@@ -0,0 +1,127 @@
+                    should be passed without quotations.
+        "parentName" is the name of the testing folder in the drive
+                and only a specific set of values are allowed.
+        The parent folder is supposedly going to be fixed for a
+        long period of time.
```
What exactly is meant by this?

youssef-m-ali replied:
I seriously don't know lol. It should have been something in an old version that we forgot to update.

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/driveClass.py:
```diff
@@ -0,0 +1,127 @@
+            lowerDictionary[key.lower()] = value
+        try:
+            self.parentFolderID = lowerDictionary[parentName.lower()]
+        except NameError:
```
except a KeyError instead of a NameError

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/driveClass.py:
```diff
@@ -0,0 +1,127 @@
+            # Uploads to a default folder if the folder name does
+            # Not exist in the dictionary
+            self.parentFolderID = self.parentDictionary['default']
+            logging.error('Folder name not found \nuploading to "Default"')
```
Again, maybe this could be a logging.warning, @colton-smith you probably have a better feel for this than me though, thoughts?

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/driveClass.py:
```diff
@@ -0,0 +1,127 @@
+            return
+        
+        # Extracts the .csv file name from the path
+        filename= os.path.basename(filePath)
```
filename = os....
Missing a space between filename and =

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/driveClass.py:
```diff
@@ -0,0 +1,127 @@
+        if response:
+            logging.warning('Uploaded "%s" (%s)' % (metadata['name'], response['mimeType']))
+        else:
+            logging.warning('error in uploading the csv file')
```
Other way around for this, I feel like this might be better suited to be logging.error

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/driveClass.py:
```diff
@@ -0,0 +1,127 @@
+        else:
+            logging.warning('error in uploading the csv file')
+        
+    def checkExistingSubFolder(self, folderName):
```
If I remember correctly this method only gets called by the execute method correct? If so, I would consider this as a "private method" of sorts. In python everything in the class is public, however there is a convention that when defining private methods, you name them with a double underscore (__) in front. So this and getFolderId could be named:
```python
def __checkExistingSubFolder(self, folderName):
    ....

def __getFolderId(self, fName):
    ...
```
This way, people utilizing this class will know that they are not supposed to directly call these methods. Though again this is very nit-picky of me haha

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/drive_options.json:
```diff
@@ -0,0 +1,9 @@
+{
+
+"Safety":"1Vnb9UOTatdZUpcBXr1LBjziwNYZzRXQz",
+
+"Integrity": "1Kyr_bJ5MSN3b6w1KMZzdJ7YPyTgPxze1",
+
+"DTS": "100OsZP0JQIk9_VXHXpK8mdhS9GOp3yKm"
```
Convention for JSON is to indent these by 2 spaces

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/uploadToDrive.py:
```diff
@@ -0,0 +1,118 @@
+from __future__ import print_function
```
I don't think this file is necessary anymore, you can remove it from git using `git rm uploadToDrive.py` and then committing that

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/driveClass.py:
```diff
@@ -0,0 +1,127 @@
+            logging.warning('error in uploading the csv file')
+        
+    def checkExistingSubFolder(self, folderName):
+        """ Queries the drive for the folder with the given name and parent folder
+        """
```
See Colton's comment on UploadToDrive.py about the single line docstring

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/driveClass.py:
```diff
@@ -0,0 +1,127 @@
+            logging.warning("More than one file found")
+
+    def getFolderId(self, fName):
+        """ Performs the query to return the ID of the folder with the given name
+        """
```
See above about single-line docstring

Burke-Daniel reviewed (changes requested):
Looking good guys, a lot of my changes are very picky/unnecessary, just wanna make sure you guys are developing good style/practices as well!

youssef-m-ali committed:
change error message to a more descriptive warning message

youssef-m-ali committed:
fixed unnecessary dumping and opening of JSON file

youssef-m-ali committed:
indent the parameters in docstring

youssef-m-ali committed:
fix multiple error/warning issues

youssef-m-ali committed:
Delete uploadToDrive.py

youssef-m-ali committed:
fix json indents

youssef-m-ali committed:
add __ in front of private methods

youssef-m-ali committed:
fix execute() docstring and dictionary's exception handling

youssef-m-ali committed:
update the tool to reflect a class implementation

youssef-m-ali committed:
Fixed a comment indentation

youssef-m-ali committed:
Improved the wording and fixed a comment

youssef-m-ali committed:
created the README.md

Detailed steps on how to use the tool included, some sections still missing and needs some proofreading

Burke-Daniel commented on Pidaq/Lib/uploadToDrive/README.md:
```diff
@@ -0,0 +1,79 @@
+4. Close the browser tab and return to the command-line. Continue from step 5 in How to use RunInCmd.py
+
+## Setting up drive_options.json
+This JSON (dictionary?) will include the names of the testing folders such as "Wind Tunnel" or "DTS" and the corresponding folder IDs.
```
Yeah JSON (Javascript Object Notation) would be equivalent to a python dict

Burke-Daniel reviewed (approved):
Great job guys! GTM once you finish the client secret part of the README

youssef-m-ali committed:
fixed JSON files naming

youssef-m-ali committed:
Update README.md

youssef-m-ali committed:
Rename RunInCmd.py to runInCmd.py

youssef-m-ali committed:
Changed location of defaultID variable

youssef-m-ali committed:
finalized edits in README

youssef-m-ali committed:
removed unnecessary hyperlinks

youssef-m-ali committed:
Delete client_secret.json

kmyali committed:
Fixed a description

colton-smith reviewed (approved):
GTM
